tests/unit/auth/oidc.ts

```diff
@@ -0,0 +1,91 @@
+import { createClient, rqlBuilder } from '../../../src';
+import { createProviderData, generateRandomName } from '../../__helpers__/oidc';
+import { OidcProviderResponse } from '../../../src/models/oidc/Providers';
+
+describe('Enables an OpenID Connect workflow', () => {
+  let provider: OidcProviderResponse;
+
+  const sdk = createClient({
+    host: process.env.API_HOST,
+    consumerKey: process.env.CONSUMER_KEY,
+    consumerSecret: process.env.CONSUMER_SECRET,
+  });
+
+  beforeAll(async () => {
+    await sdk.auth.authenticate({
+      email: process.env.API_USERNAME,
+      password: process.env.API_PASSWORD,
+    });
+  });
+
+  beforeEach(async () => {
+    const data = createProviderData();
+    provider = await sdk.auth.oidc.createProvider(data);
+  });
+
+  afterAll(async () => {
+    await sdk.auth.oidc.deleteProvider(provider.id);
+  });
+
+  it('Creates an OpenID Connect provider', async () => {
```

Comment:
I would still do this create a call in the test here, now the test and the beforeEach are mangled.